pkg/controllers/jobset_controller.go

```diff
@@ -149,6 +150,8 @@ func (r *JobSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 			return ctrl.Result{}, err
 		}
 	}
+	// Calculate PodsReady and update statuses for each ReplicatedJob
+	r.calculateAndUpdateReplicatedJobsStatuses(ctx, &js, ownedJobs)
```

Comment:
I think we probably want to allow for returning an error and trigger a reconcile.  Following a similar pattern as we do in this function.